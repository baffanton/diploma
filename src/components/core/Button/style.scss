@import 'global.scss';

.button {
    text-align: center;
    border: none;
    border-radius: 20px;
    cursor: pointer;
    box-sizing: border-box;
    font-size: 25px;
}

.button_theme_primary {
    background-color: $blue;
    color: $white;
}

.button_theme_ordinary {
    background-color: $darkBlue;
    color: $white;
}

.button_theme_white {
    background-color: $white;
    color: $darkBlue;
}

.button_with-icon {
    display: flex;
    justify-content: space-between;
}

.button__title {
    text-align: center;
    font-size: 30px;
    color: $white;
}

.button__icon {
    height: 85%;
    padding-left: 12px;
    color: $yellow;
}

.button__icon_color_primary {
    border-left: 2px solid $darkBlue;
}

.button__icon_color_ordinary,
.button__icon_color_white {
    border-left: 2px solid $blue;
}

.button_height_s {
    height: 42px;
}

.button_height_m {
    height: 60px;
}

.button_height_l {
    height: 72px;
}
