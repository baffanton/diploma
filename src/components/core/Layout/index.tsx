import React from 'react';

import { ILayout } from './types';
import cx from 'classnames';

import './style.scss';

const Layout: React.FC<ILayout> = ({ className, children, onClick }) => {
    return (
        <div className={cx('layout', className)} onClick={onClick}>
            {children}
        </div>
    );
};

export { Layout };
