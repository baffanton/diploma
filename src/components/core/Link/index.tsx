import React from 'react';

import { ILink } from './types';
import cx from 'classnames';

import './style.scss';

const Link: React.FC<ILink> = ({ children, href, className = '', onClick }) => {
    const classNames = cx('link', className);

    return (
        <a className={classNames} href={href} onClick={onClick} target="_blank" rel="noreferrer">
            {children}
        </a>
    );
};

export { Link };
