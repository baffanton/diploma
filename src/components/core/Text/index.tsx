import React from 'react';
import { IText } from './types';
import cx from 'classnames';
import { SizeEnum } from 'enums/sizeTypes';
import { StyleEnum } from 'enums/styleTypes';
import { WeightEnum } from 'enums/weightTypes';
import './style.scss';

const Text: React.FC<IText> = ({
    className,
    children,
    fontSize = SizeEnum.medium,
    fontWeight = WeightEnum.default,
    fontStyle = StyleEnum.default,
    onClick,
    pointer,
}) => {
    const classNames = cx(
        'text',
        `text_size_${fontSize}`,
        `text_style_${fontStyle}`,
        `text_weight_${fontWeight}`,
        (onClick || pointer) && 'text_cursor_pointer',
        className,
    );

    return (
        <p className={classNames} onClick={onClick}>
            {children}
        </p>
    );
};

export { Text };
