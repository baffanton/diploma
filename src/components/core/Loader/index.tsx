import React from 'react';

import { Layout } from '../Layout';

import './style.scss';

const Loader = () => {
    return (
        <Layout className="loader__wrapper">
            <span className="loader"></span>
        </Layout>
    );
};

export { Loader };
