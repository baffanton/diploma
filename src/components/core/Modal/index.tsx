import React from 'react';

import { Layout } from 'components/core/Layout';

import { getModal } from './helpers';
import { IModalProps } from './types';

import './style.scss';

const Modal: React.FC<IModalProps> = ({ type, closeModal, option }) => {
    return <Layout className="modal-window">{getModal(type, closeModal, option)}</Layout>;
};

export { Modal };
