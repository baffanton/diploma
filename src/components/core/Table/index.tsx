import React, { useMemo } from 'react';
import { useTable } from 'react-table';

import { Layout } from 'components/core/Layout';
import { Title } from 'components/core/Title';

import { ITable } from './types';
import cx from 'classnames';
import { SizeEnum } from 'enums/sizeTypes';
import { WeightEnum } from 'enums/weightTypes';

import './style.scss';

const Table: React.FC<ITable> = ({ config, tableData, isClickable = false, selectedRowIndex, setSelectedRowIndex }) => {
    const columns = useMemo(() => config, [config]);
    const data: any = useMemo(() => tableData, [tableData]);

    const tableInstance = useTable({
        columns,
        data,
    });

    if (!data.length) {
        return (
            <Layout className="table_empty">
                <Title className="table__title_empty" fontSize={SizeEnum.medium} fontWeight={WeightEnum.bold}>
                    Нет данных
                </Title>
            </Layout>
        );
    }

    const onChangeSelectedRow = (id: string) => {
        if (!isClickable || !setSelectedRowIndex) {
            return null;
        }

        if (id === selectedRowIndex) {
            return setSelectedRowIndex(null);
        }

        setSelectedRowIndex(id);
    };

    const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = tableInstance;

    return (
        <Layout className="table">
            <table className="table__thead-container" {...getTableProps()}>
                <thead className="table__thead">
                    {headerGroups.map((headerGroup) => (
                        <tr className="table__thead_tr" {...headerGroup.getHeaderGroupProps()}>
                            {headerGroup.headers.map((column) => (
                                <th className="table__thead_th" {...column.getHeaderProps()}>
                                    {column.render('Header')}
                                </th>
                            ))}
                        </tr>
                    ))}
                </thead>
            </table>
            <Layout className="table__container">
                <table className="table__tbody-container" {...getTableProps()}>
                    <tbody className="table__tbody" {...getTableBodyProps()}>
                        {rows.map((row) => {
                            prepareRow(row);
                            const checkIdentify = selectedRowIndex === row.id;

                            return (
                                <tr
                                    className={cx(
                                        'table__tbody_tr',
                                        isClickable && checkIdentify && 'table__tbody_tr_isSelected',
                                    )}
                                    {...row.getRowProps()}
                                    onClick={() => onChangeSelectedRow(row.id)}
                                >
                                    {row.cells.map((cell) => (
                                        <td className="table__tbody_td" {...cell.getCellProps()}>
                                            {cell.render('Cell')}
                                        </td>
                                    ))}
                                </tr>
                            );
                        })}
                    </tbody>
                </table>
            </Layout>
        </Layout>
    );
};

export { Table };
