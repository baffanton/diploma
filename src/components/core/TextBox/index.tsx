import React from 'react';

import { Layout } from 'components/core/Layout';

import { Icon } from '../Icon';
import { ITextBox } from './types';
import cx from 'classnames';
import { ColorThemeType } from 'enums/colorThemeTypes';
import { InputTypesEnum } from 'enums/inputTypes';
import { SizeEnum } from 'enums/sizeTypes';

import './style.scss';

const TextBox: React.FC<ITextBox> = (
    {
        id,
        name = '',
        label,
        onChange,
        classNameInput = '',
        classNameLabel = '',
        classNameContainer = '',
        placeholder = '',
        error,
        register,
        value,
        icon,
        onIconClick,
        labelPosition,
        type = InputTypesEnum.text,
        heightType = SizeEnum.medium,
        colorTheme = ColorThemeType.primary,
        disabled,
    },
    { ...props },
) => {
    const inputClassNames = cx(
        'text-box__input',
        `text-box__input_color_${colorTheme}`,
        `text-box__input_height_${heightType}`,
        classNameInput,
    );
    const labelClassNames = cx(
        'text-box__label',
        `text-box__label_color_${colorTheme}`,
        `text-box__label_height_${heightType}`,
        classNameLabel,
    );
    const containerClassNames = cx(
        'text-box__wrapper',
        labelPosition && `text-box__wrapper_label_${labelPosition}`,
        classNameContainer,
    );
    const labelContainerClassNames = cx(
        'text-box__label-container',
        `text-box__label-container_position_${labelPosition}`,
    );

    const inputRow = () => {
        if (icon) {
            return (
                <Layout className="text-box__input-container">
                    <input
                        id={id}
                        name={name}
                        className={inputClassNames}
                        placeholder={placeholder}
                        type={type}
                        onChange={onChange}
                        value={value}
                        disabled={disabled}
                        autoComplete={type === InputTypesEnum.password ? 'on' : 'off'}
                        {...register}
                    />
                    <Icon
                        className="text-box__icon"
                        onClick={onIconClick}
                        fontAwesomeIcon={icon}
                        heightType={SizeEnum.short}
                    />
                </Layout>
            );
        }

        return (
            <input
                id={id}
                name={name}
                className={inputClassNames}
                placeholder={placeholder}
                type={type}
                onChange={onChange}
                value={value}
                disabled={disabled}
                autoComplete={type === InputTypesEnum.password ? 'on' : 'off'}
                {...register}
            />
        );
    };

    return (
        <Layout className={containerClassNames}>
            <Layout className={labelContainerClassNames}>
                {label && (
                    <label className={labelClassNames} htmlFor={id}>
                        {label}
                    </label>
                )}
                {error && (
                    <p className={cx('text-box__error', `text-box__error_theme_${colorTheme}`)}>{error.message}</p>
                )}
            </Layout>
            {inputRow()}
        </Layout>
    );
};

export { TextBox };
