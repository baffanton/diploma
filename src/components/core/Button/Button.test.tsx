import React from 'react';

import { Button } from 'components/core/Button';

import { faEye } from '@fortawesome/free-solid-svg-icons';
import { render, screen } from '@testing-library/react';

describe('Button component', () => {
    it('Button renders', () => {
        render(<Button onClick={() => null} />);

        expect(screen.getByRole('button')).toBeInTheDocument();
    });

    it('Button with icon haven`t button role', () => {
        render(<Button icon={faEye} title="Eye" onClick={() => null} />);

        expect(screen.queryByRole('button')).toBeNull();
    });

    it('Button snapshot', () => {
        const button = render(<Button icon={faEye} title="Eye" onClick={() => null} />);

        expect(button).toMatchSnapshot();
    });

    it('Button with icon snapshot', () => {
        const button = render(<Button onClick={() => null} />);

        expect(button).toMatchSnapshot();
    });

    it('Button styles working', () => {
        render(<Button onClick={() => null} />);

        expect(screen.getByRole('button')).toHaveClass('button');
    });
});
