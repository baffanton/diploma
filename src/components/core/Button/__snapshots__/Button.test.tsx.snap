// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout button_with-icon button button_height_m button_theme_primary"
      >
        <p
          class="text text_size_m text_style_default text_weight_default button__title"
        >
          Eye
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-eye icon icon_heightType_s button__icon button__icon_theme_primary"
          data-icon="eye"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout button_with-icon button button_height_m button_theme_primary"
    >
      <p
        class="text text_size_m text_style_default text_weight_default button__title"
      >
        Eye
      </p>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-eye icon icon_heightType_s button__icon button__icon_theme_primary"
        data-icon="eye"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component Button with icon snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button button_height_m button_theme_primary"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="button button_height_m button_theme_primary"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
